src/3d/chunks/qgschunkedentity_p.cpp

```diff
@@ -62,6 +62,16 @@ QgsChunkedEntity::QgsChunkedEntity( float tau, QgsChunkLoaderFactory *loaderFact
   mRootNode = loaderFactory->createRootNode();
   mChunkLoaderQueue = new QgsChunkList;
   mReplacementQueue = new QgsChunkList;
+
+  int memoryAvailableKB = Qgs3DUtils::estimateGpuMemoryAvailable();
+  if ( memoryAvailableKB == -1 )
+  {
+    mGpuMemoryLimit = 500.0;
+  }
+  else
+  {
+    mGpuMemoryLimit = 0.8 * static_cast<double>( memoryAvailableKB ) / 1024.0;
+  }
```

Comment:
How about we leave the default 500mb value only defined in the header: (and add a comment next to it that it's a conservative default value for when the actual GPU available memory isn't available)
```suggestion
  if ( memoryAvailableKB != -1 )
  {
    mGpuMemoryLimit = 0.8 * static_cast<double>( memoryAvailableKB ) / 1024.0;
  }
```